### ISSUE TITLE
ProphetNetForConditionalGeneration model isn't returning all objects properly

### ISSUE THREAD

**Issue by Raian-Rahman** [huggingface/transformers#11921]
## Environment info

- `transformers` version: 4.6.1
- Platform: Google Colab
- Python version: 
- PyTorch version (GPU?): GPU
- Tensorflow version (GPU?): 
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help
- text generation: @patrickvonplaten

## Information

The model I am using Prophetnet:

The problem arises when using:
* my own modified scripts: [My notebook](https://colab.research.google.com/drive/1rmZyTXsdEDDpx8tbX-Gt6Uj9NuQi92VK?usp=sharing)

The tasks I am working on is:
*  an official SQUaD task: Question Generation

## To reproduce

1. Just run the notebook 
2. After running a single inference I am only getting 4 objects while I should get loss and other objects.

## Expected behavior

After running a single inference I am only getting 4 objects while I should get loss and other objects.  @patrickvonplaten

**Comment by LysandreJik** [comment]
What are all the objects you expect to get? The `loss` is only returned if you pass the labels to the model - otherwise it cannot compute any loss. Please check out the [return statement of ProphetNetForConditionalGeneration's forward method for more information](https://huggingface.co/transformers/model_doc/prophetnet.html#transformers.ProphetNetForConditionalGeneration.forward).

**Comment by Raian-Rahman** [comment]
Thank you, it worked @LysandreJik
